### PR TITLE
Fix #12365: Company inauguration date should use calendar year in wallclock mode

### LINKED ISSUE #12365: [Bug]: Company inauguration date should use calendar year in wallclock mode
### Version of OpenTTD

14.0-RC3

### Expected result

Company inauguration date, similar to eg. vehicle design dates, should probably always be displayed as calendar year.

### Actual result

image

In wallclock mode, inauguration time is in minutes(?) with no units, and looks quite weird.

### Steps to reproduce

1. Set the time setting to wallclock mode
2. Start a new game and look at the company information window

### PR DESCRIPTION
**Pull request by Vinixss**
## Motivation / Problem
When using the Wallclock timekeeping setting, the Inauguration year of the company is displayed in minutes rather than the actual calendar year

Closes #12365. (edit 2TT)

## Description
Added a new variable to company_base.h named "inaugurated_year_calendar" to store the calendar time version of the inaugurated year. Both this and the economic inaugurated_year are passed as parameters into the company window in company_gui.cpp in order to display both.

## Limitations
Main limitation is that, in order to fix the bug in the way described under Issue #12365, I had to edit english.txt and add a new value, so other translation files do not have this

## Checklist for review
Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* This PR touched english.txt

### CODE REVIEW

File: src/lang/english.txt [hunk 1]
```diff
@@ -3886,7 +3886,7 @@ STR_FINANCES_INFRASTRUCTURE_BUTTON                              :{BLACK}Infrastr
 STR_COMPANY_VIEW_CAPTION                                        :{WHITE}{COMPANY} {BLACK}{COMPANY_NUM}
 STR_COMPANY_VIEW_PRESIDENT_MANAGER_TITLE                        :{WHITE}{PRESIDENT_NAME}{}{GOLD}(Manager)
 
-STR_COMPANY_VIEW_INAUGURATED_TITLE                              :{GOLD}Inaugurated: {WHITE}{NUM}
+STR_COMPANY_VIEW_INAUGURATED_TITLE                              :{GOLD}Inaugurated: {WHITE}{NUM} (period {NUM})
```
Comment (James103) [inline src/lang/english.txt]: This should be split into two strings: one containing just `{GOLD}Inaugurated: {WHITE}{NUM}` for calendar timekeeping, and the new string `{GOLD}Inaugurated: {WHITE}{NUM} (period {NUM})` for wallclock timekeeping.

Comment (Vinixss) [reply src/lang/english.txt]: Thank you, will do that

File: src/company_base.h [hunk 2]
```diff
@@ -92,6 +92,7 @@ struct CompanyProperties {
 	TileIndex last_build_coordinate; ///< Coordinate of the last build thing by this company.
 
 	TimerGameEconomy::Year inaugurated_year; ///< Economy year of starting the company.
+    TimerGameCalendar::Year inaugurated_year_calendar; ///< Calendar year of starting the company. Used for when we need to display an actual Inaguration Year while in Wallclock mode.
```
Comment (2TallTyler) [inline src/company_base.h]:
```suggestion
    TimerGameCalendar::Year inaugurated_year_calendar; ///< Calendar year of starting the company. Used for when we need to display an actual Inauguration Year while in Wallclock mode.
```

**Review by 2TallTyler — COMMENTED** [review]
Thank you for the contribution! You will need to clean up your commit history with an [interactive rebase](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history#interactive-rebase) and then force-push. Feel free to ask if you have any questions!

File: src/company_gui.cpp [hunk 3]
```diff
@@ -2400,9 +2404,13 @@ struct CompanyWindow : Window
...
 
-			case WID_C_DESC_INAUGURATION:
-				SetDParam(0, Company::Get((CompanyID)this->window_number)->inaugurated_year);
-				break;
+            case WID_C_DESC_INAUGURATION:
+                if (TimerGameEconomy::UsingWallclockUnits()) {
+                    SetDParam(0, Company::Get((CompanyID) this->window_number)->inaugurated_year_calendar);
+                    SetDParam(1, Company::Get((CompanyID) this->window_number)->inaugurated_year);
+                }
+                else SetDParam(0, Company::Get((CompanyID) this->window_number)->inaugurated_year);
```
Comment (2TallTyler) [inline src/company_gui.cpp:2412]: `else` goes on the same line as the closing bracket for the `if`, and also needs brackets. See my string parameters snippet above for the correct syntax.

File: src/company_base.h [hunk 4]
```diff
@@ -92,6 +92,7 @@ struct CompanyProperties {
 	TileIndex last_build_coordinate; ///< Coordinate of the last build thing by this company.
 
 	TimerGameEconomy::Year inaugurated_year; ///< Economy year of starting the company.
+    TimerGameCalendar::Year inaugurated_year_calendar; ///< Calendar year of starting the company. Used for when we need to display an actual Inauguration Year while in Wallclock mode.
```
Comment (2TallTyler) [inline src/company_base.h:95]: We use tabs for indentation, not spaces.

File: src/company_gui.cpp [hunk 5]
```diff
@@ -2237,6 +2237,10 @@ struct CompanyWindow : Window
 			}
 		}
 
+        if (TimerGameEconomy::UsingWallclockUnits()) {
+            this->GetWidget<NWidgetLeaf>(WID_C_DESC_INAUGURATION)->SetDataTip(STR_COMPANY_VIEW_INAUGURATED_TITLE_WALLCLOCK, STR_NULL);
+        }
+
```
Comment (2TallTyler) [inline src/company_gui.cpp:2243]: Good try, but we typically use a different way to change strings in GUI.

First, in the widget structure use a string that allows us to select strings as a parameter. In this case, you need to set two string parameters, so it should be STR_JUST_STRING2, like so:

```
NWidget(WWT_TEXT, COLOUR_GREY, WID_C_DESC_INAUGURATION), SetDataTip(STR_JUST_STRING2, STR_NULL), SetFill(1, 0),
```

Then choose the correct string in SetStringParameters:

```
case WID_C_DESC_INAUGURATION:
    if (TimerGameEconomy::UsingWallclockUnits()) {
		SetDParam(0, STR_COMPANY_VIEW_INAUGURATED_TITLE_WALLCLOCK);
		SetDParam(1, Company::Get((CompanyID) this->window_number)->inaugurated_year_calendar);
        SetDParam(2, Company::Get((CompanyID) this->window_number)->inaugurated_year);
	} else {
		SetDParam(0, STR_COMPANY_VIEW_INAUGURATED_TITLE);
		SetDParam(1, Company::Get((CompanyID)this->window_number)->inaugurated_year);
	}
    break;
```

File: src/company_cmd.cpp [hunk 6]
```diff
@@ -608,6 +608,7 @@ Company *DoStartupNewCompany(bool is_ai, CompanyID company = INVALID_COMPANY)
 	c->avail_railtypes = GetCompanyRailTypes(c->index);
 	c->avail_roadtypes = GetCompanyRoadTypes(c->index);
 	c->inaugurated_year = TimerGameEconomy::year;
+    c->inaugurated_year_calendar = TimerGameCalendar::year;
```
Comment (2TallTyler) [inline src/company_cmd.cpp:611]: Intent with tabs, not spaces.

**Review by 2TallTyler — COMMENTED** [review]
You will need to add savegame conversion, so when somebody loads a savegame created before this data was saved, something will show up. I'm not sure what to show here, but here's how you'd do it: First, create a new SaveLoadVersion entry in `saveload.h`, like so:

```
SLV_COMPANY_INAUGURATED_PERIOD,         ///< 337  PR#12769 Companies show the period inaugurated in wallclock mode.
```
Then add the conversion code in `afterload.cpp`, like so:
```
if (IsSavegameVersionBefore(SLV_COMPANY_INAUGURATED_PERIOD)) {
	for (Company *c : Company::Iterate()) {
		c->inaugurated_year_calendar = (whatever value you want to set)
	}
}
```
You will need to squash all your commits into one in order for CI to succeed, otherwise the commit checker will fail.

Also, stop clicking the `Update branch` button on GitHub: it adds merge commits and makes a mess of your PR. As long as the check says `Merging can be performed automatically with 1 approving review.`, you do not need to update anything. 😃

**Comment by 2TallTyler** [comment]
You might find this branch helpful as a starting point: https://github.com/2TallTyler/OpenTTD/tree/inauguration-demonstration

I am not trying to replace your work, just wanted to test my suggestions to ensure I'm not leading you astray! 😄

**Comment by Vinixss** [comment]
Thank you. Sorry about the awkward commits, this is my first time using a rebase so I'm trying to fix that

**Pull request closed without merging**